src/function/gds/output_writer.cpp

```diff
@@ -96,6 +98,10 @@ void PathsOutputWriter::write(processor::FactorizedTable& fTable, nodeID_t dstNo
                 beginWritePath(0);
             }
             fTable.append(vectors);
+            // No need to check against limit number since this is the first output.
+            if (counter != nullptr) {
+                counter->increase(1);
```

Comment:
I don't understand this comment. What if the limit is 1?

Reply:
Then we will continue and terminate when we find the second path. There is a LIMIT operator at the very end which guarantees we will not exceed 1.